### PR TITLE
Add view hierarchy docs for react native

### PR DESCRIPTION
**Pull request by krystofwoldrich**
- https://github.com/getsentry/sentry-react-native/pull/2708

### CODE REVIEW

File: src/platforms/common/enriching-events/viewhierarchy.mdx [hunk 1]
```diff
@@ -29,3 +30,49 @@ View hierarchy debugging is an opt-in feature. You can enable it as shown below:
...
 
 ![Screenshots List Example](viewhierarchy-list-example.png)
+
+<PlatformSection supported={["react-native"]}>
+
+## View Hierarchy mapping to React Native components
+
+View Hierarchy captured by the SDK represents the native user interface generated based on React Native components structure. Generally each React Element maps to a native component, but there are exception like [View Flattening](https://reactnative.dev/architecture/view-flattening) described in the React Native documentation.
+
+Part of View Hierarchy are also native components wrapping the app created by React Native.
```
Comment (lizokm) [inline src/platforms/common/enriching-events/viewhierarchy.mdx:40]:
```suggestion
Native components wrapping the app (created by React Native), are another part of View Hierarchy.
```

File: src/platforms/common/enriching-events/viewhierarchy.mdx [hunk 2]
```diff
@@ -29,3 +30,49 @@ View hierarchy debugging is an opt-in feature. You can enable it as shown below:
...
+
+### iOS
+
+The code snipped below show React Native Component and next to it native UIKit components as they will be included in the view hierarchy.
+
```
Comment (lizokm) [inline src/platforms/common/enriching-events/viewhierarchy.mdx:45]:
```suggestion
The code snippet below shows React Native and native UIKit components the way they would be included in the view hierarchy.

```

File: src/platforms/common/enriching-events/viewhierarchy.mdx [hunk 3]
```diff
@@ -29,3 +30,49 @@ View hierarchy debugging is an opt-in feature. You can enable it as shown below:
...
+      <Text>Hello World!</Text>  // RCTTextView
+    </View>
+  );
+}
+```
+
+### Android
+
+The code snipped below show React Native Component and next to it native Android components as they will be included in the view hierarchy.
+
```
Comment (lizokm) [inline src/platforms/common/enriching-events/viewhierarchy.mdx:62]:
```suggestion
The code snippet below shows React Native and native Android components the way they would be included in the view hierarchy.

```

**Review by lizokm — APPROVED** [review]
Made some tweaks.

**Comment by krystofwoldrich** [comment]
> Made some tweaks.

Thanks.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**